File: JavaScript/Math-Array-String/40.js
// 1678. Goal Parser Interpretation


/**
 * @param {string} command
 * @return {string}
 */
var interpret = function(command) {
    // Replace "()" with "o"
    command = command.replace(/\(\)/g, "o");
    // Replace "(al)" with "al"
    command = command.replace(/\(al\)/g, "al");
    return command;
};

// Examples
console.log(interpret("G()(al)")); // Output: "Goal"
console.log(interpret("G()()()()(al)")); // Output: "Gooooal"

File: Python/Math-Array-String/40.py
# 1678. Goal Parser Interpretation


class Solution:
    def interpret(self, command: str) -> str:
        # Replace "()" with "o"
        command = command.replace("()", "o")
        # Replace "(al)" with "al"
        command = command.replace("(al)", "al")
        return command
    

sol = Solution()
output = sol.interpret("G()(al)")
print(output)